runtime/commonMain/src/kotlinx/serialization/json/JsonTransformingSerializer.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlinx.serialization.*
+import kotlinx.serialization.internal.*
+import kotlinx.serialization.json.internal.*
+
+/**
+ * Base class for custom serializers that allows manipulating an abstract JSON
+ * representation of the class.
+ *
+ * Instead of direct interaction with [Encoder] or [Decoder], this class
```

Comment:
less 'you', more 'we': `this class allows manipulations with [JsonElement] directly`.